org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -146,9 +146,15 @@ private static Diagnostic toDiagnostic(Range range, IMarker marker) {
 		}
 		Diagnostic d = new Diagnostic();
 		d.setSource(JavaLanguageServerPlugin.SERVER_SOURCE_ID);
-		d.setMessage(marker.getAttribute(IMarker.MESSAGE, ""));
+		String message = marker.getAttribute(IMarker.MESSAGE, "");
+		if (message != null && message.startsWith("Project configuration is not up-to-date with pom.xml.")) {
+			message = "Project configuration is not up-to-date with pom.xml, requires an update.";
+			d.setSeverity(DiagnosticSeverity.Warning);
```

Comment:
No the severity change should be at the m2e preferences level, or else the build command will continue to fail if such an error is reported. See https://github.com/Microsoft/vscode-java-debug/issues/358